daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -543,179 +540,7 @@ codegenTests codegenDir damlTypes = testGroup "daml codegen" (
-                writeFileUTF8 tokenFile ("Bearer " <> makeSignedJwt sharedSecret <> "\n")
-                withCreateProcess sandboxProc  $ \_ _ _ ph ->
-                    race_ (waitForProcess' sandboxProc ph) $ do
-                        waitForConnectionOnPort (threadDelay 100000) port
-                        withCurrentDirectory (deployDir </> "proj2") $ do
-                            callCommandQuiet $ unwords
-                                [ "daml deploy"
-                                , "--access-token-file " <> tokenFile
-                                , "--port", show port
-                                , "--host localhost"
-                                ]
-                        -- waitForProcess' will block on Windows so we explicitly kill the process.
-                        terminateProcess ph
-
-makeSignedJwt :: String -> String
-makeSignedJwt sharedSecret = do
-  let urc = JWT.ClaimsMap $ Map.fromList [ ("admin", Aeson.Bool True)]
-  let cs = mempty { JWT.unregisteredClaims = urc }
-  let key = JWT.hmacSecret $ T.pack sharedSecret
-  let text = JWT.encodeSigned key mempty cs
-  T.unpack text
-
-
 damlInstallerName :: String
 damlInstallerName
     | isWindows = "daml.exe"
     | otherwise = "daml"
-
--- | Like call process but returning stdout.
-runCreateProcessForStdout :: CreateProcess -> IO String
-runCreateProcessForStdout createProcess = do
-    -- We use `repeat ' '` to keep stdin open. Really we would just
-    -- like to inherit stdin but readCreateProcessWithExitCode does
-    -- not allow us to overwrite just that and I don’t want to
-    -- reimplement everything.
-    (exit, out, err) <- readCreateProcessWithExitCode createProcess (repeat ' ')
-    hPutStr stderr err
-    unless (exit == ExitSuccess) $ throwIO $ ProcessExitFailure exit createProcess
-    return out
-
-callCommandForStdout :: String -> IO String
-callCommandForStdout cmd =
-    runCreateProcessForStdout (shell cmd)
-
--- | Like call process but hiding stdout.
-runCreateProcessQuiet :: CreateProcess -> IO ()
-runCreateProcessQuiet createProcess = do
-  _ <- runCreateProcessForStdout createProcess
-  return ()
-
--- | Like callProcess but hides stdout.
-callProcessQuiet :: FilePath -> [String] -> IO ()
-callProcessQuiet cmd args =
-    runCreateProcessQuiet (proc cmd args)
-
--- | Like callCommand but hides stdout.
-callCommandQuiet :: String -> IO ()
-callCommandQuiet cmd =
-    runCreateProcessQuiet (shell cmd)
-
-data ProcessExitFailure = ProcessExitFailure !ExitCode !CreateProcess
-    deriving (Show, Typeable)
-
-instance Exception ProcessExitFailure
-
--- This is slightly hacky: we need to find a free port but pass it to an
--- external process. Technically this port could be reused between us
--- getting it from the kernel and the external process listening
--- on that port but ports are usually not reused aggressively so this should
--- be fine and is certainly better than hardcoding the port.
-getFreePort :: IO PortNumber
```

Comment:
I’d like to keep this here. This is a rather gross hack and I don’t want to use it anywhere else.

Reply:
The reason for moving the `getFreePort` was because at one point during the refactor I needed to use it in two different test suites. In general I don't think it is a good idea to try to hide hackiness by obscuring it. But anyway I moved the `FreePort.hs` file back into the integration-tests, so we don't encourage it's further use.